crates/agentgateway/src/types/local.rs

```diff
@@ -229,9 +236,16 @@ impl LocalBackend {
 	}
 }
 
+#[apply(schema!)]
+pub enum McpStatefulMode {
+	Stateless,
+	Stateful,
+}
+
 #[apply(schema!)]
 pub struct LocalMcpBackend {
 	pub targets: Vec<Arc<LocalMcpTarget>>,
+	pub stateful_mode: Option<McpStatefulMode>
```

Comment:
doesn't really matter but fwiw you can also do.

```
#[serde(default)]
pub stateful_mode: McpStatefulMode
```

identical behavior and not important for this code but nice for some more complex cases